record/record_test.go

```diff
@@ -636,3 +642,128 @@ func TestRecoverLastCompleteBlock(t *testing.T) {
 		t.Fatalf("verifyLastBlockRecover: %v", err)
 	}
 }
+
+func TestSeekRecord(t *testing.T) {
+	recs, err := makeTestRecords(
+		// The first record will consume 3 entire blocks but a fraction of the 4th.
+		blockSize*3,
+		// The second record will completely fill the remainder of the 4th block.
+		3*(blockSize-headerSize)-2*blockSize-2*headerSize,
+		// Consume the entirety of the 5th block.
+		blockSize-headerSize,
+		// Consume the entirety of the 6th block.
+		blockSize-headerSize,
+		// Consume roughly half of the 7th block.
+		blockSize/2,
+	)
+	if err != nil {
+		t.Fatalf("makeTestRecords: %v", err)
+	}
+
+	r := NewReader(bytes.NewReader(recs.buf))
+	// Seek to a valid block offset, but within a multiblock record. This should cause the next call to
+	// Next after SeekRecord to return the next valid FIRST/FULL chunk of the subsequent record.
+	err = r.SeekRecord(blockSize)
+	if err != nil {
+		t.Fatalf("SeekRecord: %v", err)
+	}
+	rec, err := r.Next()
+	if err != nil {
+		t.Fatalf("Next: %v", err)
+	}
+	rData, _ := ioutil.ReadAll(rec)
+	if !bytes.Equal(rData, recs.records[1]) {
+		t.Fatalf("Unexpected output in record 1's data, got %v want %v", rData, recs.records[1])
+	}
+
+	// Seek to 3 bytes into the second block, which is still in the middle of the first record, but not
+	// at a valid chunk boundary.
+	err = r.SeekRecord(blockSize + 3)
+	if err != nil {
+		t.Fatalf("SeekRecord: %v", err)
+	}
+
+	if _, err = r.Next(); err == nil {
+		t.Fatalf("Unexpected lack of error calling SeekRecord to a bad offset")
+	}
+	r.Recover() // Recover from the faux error just generated.
+
+	// Seek to the fifth block and verify all records can be read as appropriate.
+	err = r.SeekRecord(blockSize * 4)
+	if err != nil {
+		t.Fatalf("SeekRecord: %v", err)
+	}
+
+	recLooper := func(i int) {
```

Comment:
I actually went with the sentinel approach. Because it's possible that calling Flush() will increment the seq number (or any other operation), and then we'd end up with LastRecordOffset returning a fairly undefined value. 

I've added a test for all these cases.